### PR TITLE
CI/build: Update Go version, linters and fix minor issues

### PR DESCRIPTION
**Pull request by thrasher-**
# PR Description

Bumps the CI and toolchain go versions to 1.23, golangci-lint to v1.60.1 and fixes issues along the way. 

Adds the following new linters:

- int range
- copyloopvar
- govet: nilness and unusedwrite to match default gopls behaviour

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: backtester/funding/funding_test.go [hunk 1]
```diff
@@ -234,20 +230,7 @@ func TestExists(t *testing.T) {
...
-		reserved:          quoteItem.reserved,
-		transferFee:       quoteItem.transferFee,
-		pairedWith:        quoteItem.pairedWith,
-		trackingCandles:   quoteItem.trackingCandles,
-		snapshot:          quoteItem.snapshot,
-		isCollateral:      quoteItem.isCollateral,
-		collateralCandles: quoteItem.collateralCandles,
-	}
+	quoteCopy := Item{pairedWith: quoteItem.pairedWith}
 	quoteCopy.pairedWith = &baseCopy
```
Comment (shazbert) [inline backtester/funding/funding_test.go:234]: `quoteCopy` can be removed as its  not being used

File: cmd/gen_otp/otp_gen.go [hunk 2]
```diff
@@ -39,7 +39,10 @@ func main() {
 		var input string
 		for {
 			log.Println("Please enter in your OTP secret:")
-			fmt.Scanln(&input)
+			if _, err = fmt.Scanln(&input); err != nil {
+				log.Printf("Failed to read input. Err: %s", err)
```
Comment (shazbert) [inline cmd/gen_otp/otp_gen.go:43]: add `\n`

**Review by shazbert — CHANGES_REQUESTED** [review]
Nice🔥

**Review by shazbert — APPROVED** [review]
tACK. Lookin' schmick.

**Review by gloriousCode — APPROVED** [review]
tACK, found some good ones in there

**Pull request merged**